zcash_client_backend/src/data_api.rs

```diff
@@ -290,22 +286,46 @@ impl AccountBalance {
     }
 
     /// Returns the total value of unspent transparent transaction outputs belonging to the wallet.
+    #[deprecated(
+        note = "this function is deprecated. Please use AccountBalance::unshielded_balance instead."
+    )]
     pub fn unshielded(&self) -> NonNegativeAmount {
-        self.unshielded
+        self.unshielded_balance.total()
+    }
+
+    /// Returns the [`Balance`] of unshielded funds in the account.
+    pub fn unshielded_balance(&self) -> &Balance {
+        &self.unshielded_balance
+    }
+
+    /// Provides a `mutable reference to the [`Balance`] of transparent funds in the account
+    /// to the specified callback, checking invariants after the callback's action has been
+    /// evaluated.
+    pub fn with_unshielded_balance_mut<A, E: From<BalanceError>>(
+        &mut self,
+        f: impl FnOnce(&mut Balance) -> Result<A, E>,
+    ) -> Result<A, E> {
+        let result = f(&mut self.unshielded_balance)?;
+        self.check_total()?;
+        Ok(result)
     }
 
     /// Adds the specified value to the unshielded total, checking for overflow of
     /// the total account balance.
+    #[deprecated(
+        note = "this function is deprecated. Please use the `Balance::add_spendable_value` on the unshielded field instead instead."
+    )]
     pub fn add_unshielded_value(&mut self, value: NonNegativeAmount) -> Result<(), BalanceError> {
-        self.unshielded = (self.unshielded + value).ok_or(BalanceError::Overflow)?;
-        self.check_total()?;
+        self.unshielded_balance.add_pending_spendable_value(value)?;
         Ok(())
     }
 
```

Comment:
This has a bug: it does not check the overflow invariant.

Option 1 (as above):
```suggestion
        self.with_unshielded_balance_mut(|unshielded_balance| {
            unshielded_balance.add_pending_spendable_value(value)
        })
    }

```

Option 2 (as above):
```suggestion
```